core/templates/dev/head/pages/exploration_editor/RouterServices.js

```diff
@@ -44,14 +44,20 @@ oppia.factory('routerService', [
       return $location.path();
     }, function(newPath, oldPath) {
       if (newPath === '') {
-        $location.path(oldPath);
+        if (oldPath === '/') {
+          $window.history.go(-1);
+        } else {
+          $location.path(oldPath);
+        }
         return;
       }
-
       if (!oldPath) {
         // This can happen when clicking on links whose href is "#".
         return;
       }
+      if (newPath === '/' && oldPath.includes('/gui/')) {
```

Comment:
The reason for using this condition is because whenever the exploration card is clicked in the dashboard page, it is first redirected to the url /create/<[exploration.id]> and then it goes to the url /create/<[exploration.id]>/gui/introduction. Now in chrome pressing the back button works correctly but in  firefox the page is redirected to the /create/<[exploration.id]> url and then it gets again redirected to the introduction page and not the previous url and hence the back button does not work correctly.